src/defaults/gitignore.txt

```diff
@@ -1,24 +1,426 @@
-# Folders to ignore - files may be too large, too many, etc.
+################################################################################
+#                          DrWatson Project Structure                          #
+################################################################################
+
+# Folders to ignore - files may be too large, too many, etc. NOTE: EDIT IF NEEDED.
 data
 videos
 plots
 notebooks
 _research
 
-# LaTeX build artifacts
-*.pdf
+################################################################################
+#                                    Julia                                     #
+################################################################################
+
+# Files generated by invoking Julia with --code-coverage
+*.jl.cov
+*.jl.*.cov
+
+# Files generated by invoking Julia with --track-allocation
+*.jl.mem
+
+# System-specific files and directories generated by the BinaryProvider and
+# BinDeps packages.
+# They contain absolute paths, and so should not be committed
+deps/deps.jl
+deps/build.log
+deps/downloads/
+deps/usr/
+deps/src/
+
+# Build artifacts for creating documentation generated by the Documenter package
+docs/build/
+docs/site/
+
+# File generated by Pkg, the package manager, based on a corresponding
+# `Project.toml`.  It records a fixed state of all packages used by the project.
+# As such, it should not be committed for packages, but should be committed for
+# applications that require a static environment.
+Manifest.toml
+
+################################################################################
+#                               Jupyter Notebook                               #
+################################################################################
+
+.ipynb_checkpoints
+*/.ipynb_checkpoints/*
+
+# Remove previous ipynb_checkpoints
```

Comment:
What do these comments here mean? Are they necessary?

Reply:
No, not really - it's just a command to remove all `.ipynb_checkpoints` from git history.  It's useful if you're using DrWatson on a previously initialized git repo, but other than that there's not a need for it.

Reply:
I think this comment then doesn't really belong in this `.gitignore` file.

Reply:
Made the change, thanks!